guarddog/analyzer/metadata/npm/npm_metadata_mismatch.py

```diff
@@ -34,46 +36,73 @@ def detect(self, package_info, path: Optional[str] = None, name: Optional[str] =
         if path is None:
             raise ValueError("path is needed to run heuristic " + self.get_name())
         package_json = Path(path) / "package" / "package.json"
-        package_manifest: dict[str, Any] = json.loads(package_json.read_text())
+        package_manifest: Dict[str, Any] = json.loads(package_json.read_text())
 
         # Get NPM manifest for version
         version_info = package_info["versions"][version]
 
-        diff: dict[str, Diff] = {
+        diff = {
             field: difference_at_key(version_info, package_manifest, field, field_type)
             for field, field_type in MANIFEST_FIELDS_CHECKLIST.items()
         }
-        number_different = sum(len(v) for k, v in diff.items())
-        diff_description = describe_diff(diff) if number_different != 0 else "No differences found"
+
+        number_different = sum(len(v) for v in diff.values())
+
+        diff_description = (
+            describe_diff(diff) if number_different != 0 else "No differences found"
+        )
+
         return number_different != 0, diff_description
 
 
 # PerItemDiff is (key, left_value, right_value)
-PerItemDiff = tuple[str, Optional[str], Optional[str]]
+PerItemDiff = tuple[str, Optional[Any], Optional[Any]]
 Diff = list[PerItemDiff]
 
 
-def diff_at_key_dict(version_at_key: dict[str, Optional[str]], manifest_at_key: dict[str, Optional[str]]) -> Diff:
+def diff_at_key_dict(
+    version_at_key: Dict[str, Optional[Any]],
+    manifest_at_key: Dict[str, Optional[Any]],
+) -> Diff:
     return [
         (key, version_at_key.get(key), manifest_at_key.get(key))
         for key in set(version_at_key.keys()).union(set(manifest_at_key.keys()))
         if version_at_key.get(key) != manifest_at_key.get(key)
     ]
 
 
-def difference_at_key(version_info: dict[str, Any], package_manifest: dict[str, Any], key: str, key_type) -> Diff:
+def difference_at_key(
+    version_info: Dict[str, Any],
+    package_manifest: Dict[str, Any],
+    key: str,
+    key_type: type,
+) -> Diff:
     version_at_key = version_info.get(key, key_type())
     manifest_at_key = package_manifest.get(key, key_type())
-    if not (isinstance(version_at_key, key_type) and isinstance(manifest_at_key, key_type)):
-        return [(f"Expected type {str(key_type)}", f"{type(version_at_key)}", f"{type(manifest_at_key)}")]
+
+    if not (
+        isinstance(version_at_key, key_type) and isinstance(manifest_at_key, key_type)
+    ):
+        return [
+            (
+                f"Expected type {str(key_type)}",
+                f"{type(version_at_key)}",
+                f"{type(manifest_at_key)}",
+            )
+        ]
     elif key_type == dict:
-        return diff_at_key_dict(version_at_key, manifest_at_key)
+        return diff_at_key_dict(version_at_key, manifest_at_key)  # type: ignore
     else:
```

Comment:
This else can be removed leaving the final return out of the if entirely (is redundant)
```suggestion
    else:
```

Reply:
Addressed in [0ec5aed](https://github.com/DataDog/guarddog/pull/331/commits/0ec5aed45cbc4a82fc5fc207c343e2f2b4513937)